core/field_colour.js

```diff
@@ -160,25 +218,54 @@ Blockly.FieldColour.activateEyedropper_ = null;
 Blockly.FieldColour.EYEDROPPER_PATH = 'eyedropper.svg';
 
 /**
- * Set a custom colour grid for this field.
- * @param {Array.<string>} colours Array of colours for this block,
- *     or null to use default (Blockly.FieldColour.COLOURS).
- * @return {!Blockly.FieldColour} Returns itself (for method chaining).
+ * Create label and readout DOM elements, returning the readout
+ * @param {string} labelText - Text for the label
+ * @return {Array} The container node and the readout node.
+ * @private
  */
-Blockly.FieldColour.prototype.setColours = function(colours) {
-  this.colours_ = colours;
-  return this;
+Blockly.FieldColour.prototype.createLabelDom_ = function(labelText) {
+  var labelContainer = document.createElement('div');
+  labelContainer.setAttribute('class', 'scratchColorPickerLabel');
+  var readout = document.createElement('span');
+  readout.setAttribute('class', 'scratchColorPickerReadout');
+  var label = document.createElement('span');
+  label.setAttribute('class', 'scratchColorPickerLabelText');
+  label.innerHTML = labelText;
```

Comment:
We have avoid using innerHtml since technically you can put whatever malicious bits you want in there. As the dom gets built up, it can become difficult to track these down. Would you mind changing your uses out here and above? I haven't tracked down all the types of elements you're calling this on, but setting the textContent should work for spans at least